zebra/zebra_rnh.c

```diff
@@ -514,23 +514,9 @@ static bool rnh_check_re_nexthops(const struct route_entry *re,
 	}
 
 	/* Some special checks if registration asked for them. */
-	if (CHECK_FLAG(rnh->flags, ZEBRA_NHT_CONNECTED)) {
-		if ((re->type == ZEBRA_ROUTE_CONNECT)
-		    || (re->type == ZEBRA_ROUTE_STATIC))
```

Comment:
This is the one thing it looks like we'd be losing with this change - we have considered STATIC as valid for reachability, and it might be disruptive to change that?

Reply:
I agree with Mark, let's just allow static routes.

Reply:
Static routes were only allowed in that one path, not the others, so it was already non-functional. And a static route with a nexthop really shouldn't be eligible to resolve a non-recursive nexthop over, no?

Reply:
(I haven't tried this but if my understanding is correct, the old code would trigger a bug here: NHT would use a static route with nexthop, send it to e.g. bgpd, bgpd would then try to install a route using that nexthop, which would be rejected because the RIB doesn't allow that same nexthop…)

Reply:
the static route would work if BGP was asking for recursive, though?

Reply:
If BGP is asking for recursive, this change shouldn't have any effect - if it does, that's a(nother) bug.